lib/rom/command.rb

```diff
@@ -140,11 +170,33 @@ def new(new_relation)
       self.class.build(new_relation, options.merge(source: relation))
     end
 
+    # @api public
+    def before(*hooks)
+      self.class.new(relation, options.merge(before: hooks))
+    end
+
+    # @api public
+    def after(*hooks)
+      self.class.new(relation, options.merge(after: hooks))
+    end
+
     private
 
     # @api private
     def composite_class
       Command::Composite
     end
+
+    # @api private
+    def apply_hooks(hooks, tuples, *args)
+      hooks.reduce(tuples) do |a, e|
+        if e.is_a?(Hash)
+          hook_meth, hook_args = e.to_a.flatten
```

Comment:
As we saw recently, `flatten(1)` is more predictable, it's also 20% faster! :grimacing: